test/unit/repositories/test_system.py

```diff
@@ -47,756 +47,764 @@
-    a duplicate System, and creates the System
-    """
-    # pylint: disable=duplicate-code
-    system_in = SystemIn(
-        **{
-            **SYSTEM_A_INFO,
-            "parent_id": None,
-        }
-    )
-    system_info = system_in.model_dump()
-    system_out = SystemOut(id=str(ObjectId()), **system_info)
-    session = MagicMock()
-    # pylint: enable=duplicate-code
-
-    # Mock `find_one` to return no duplicate system found in parent system
-    test_helpers.mock_find_one(database_mock.systems, None)
-    # Mock `insert_one` to return an object for the inserted system document
-    test_helpers.mock_insert_one(database_mock.systems, CustomObjectId(system_out.id))
-    # Mock `find_one` to return the inserted system document
-    test_helpers.mock_find_one(
-        database_mock.systems,
-        {**system_info, "_id": CustomObjectId(system_out.id)},
-    )
-
-    created_system = system_repository.create(system_in, session=session)
-
-    database_mock.systems.insert_one.assert_called_once_with(system_info, session=session)
-    assert created_system == system_out
-
-
-def test_create_with_parent_id(test_helpers, database_mock, system_repository):
-    """
-    Test creating a System with a parent ID
-
-    Verify that the `create` method properly handles the creation of a System with a parent ID
-    """
-    # pylint: disable=duplicate-code
-    system_in = SystemIn(
-        **{
-            **SYSTEM_A_INFO,
-            "parent_id": str(ObjectId()),
-        }
-    )
-    system_info = system_in.model_dump()
-    system_out = SystemOut(id=str(ObjectId()), **system_info)
-    session = MagicMock()
-
-    # Mock `find_one` to return the parent system document
-    test_helpers.mock_find_one(
-        database_mock.systems,
-        {**system_info, "_id": CustomObjectId(system_out.parent_id), "parent_id": None},
-    )
-    # pylint: enable=duplicate-code
-    # Mock `find_one` to return no duplicate system found in parent system
-    test_helpers.mock_find_one(database_mock.systems, None)
-    # Mock `insert_one` to return an object for the inserted system document
-    test_helpers.mock_insert_one(database_mock.systems, CustomObjectId(system_out.id))
-    # Mock `find_one` to return the inserted system document
-    test_helpers.mock_find_one(
-        database_mock.systems,
-        {**system_info, "_id": CustomObjectId(system_out.id)},
-    )
-
-    created_system = system_repository.create(system_in, session=session)
-
-    database_mock.systems.insert_one.assert_called_once_with(
-        {**system_info, "parent_id": CustomObjectId(system_out.parent_id)}, session=session
-    )
-    database_mock.systems.find_one.assert_has_calls(
-        [
-            call({"_id": CustomObjectId(system_out.parent_id)}, session=session),
+class SystemRepoDSL:
+    """Base class for SystemRepo unit tests"""
+
+    test_helpers: RepositoryTestHelpers
```

Comment:
It might be worth replacing RepositoryTestHelpers with a base class that the same methods, it would save adding fixtures in a setup function?

Reply:
As discussed, this can be done in a separate PR.